programs/castle-vault/src/asset_container/reserves.rs

```diff
@@ -47,21 +41,27 @@ impl AssetContainer<Reserves> {
                 },
             )
             .map(|(r, _)| r)
+            .map_err(|e| e.into())
     }
 
-    fn calculate_weights_equal(&self) -> Result<AssetContainer<Rate>, ProgramError> {
+    fn calculate_weights_equal(&self) -> Result<AssetContainer<Rate>> {
         u8::try_from(self.len())
-            .map_err(|_| ErrorCode::StrategyError.into())
-            .and_then(|num_assets| Rate::from_percent(num_assets).try_mul(100))
-            .and_then(|r| Rate::one().try_div(r))
+            .map_err(|_| ErrorCode::StrategyError)
+            .and_then(|num_assets| {
+                Rate::from_percent(num_assets)
+                    .try_mul(100)
+                    .map_err(|_| ErrorCode::StrategyError)
```

Comment:
Why are all the extra `map_err`s necessary?

Reply:
converting ProgramErrors to ErrorCodes iirc since functions now return Result<_, Error>